src/core/packages/root/server-internal/src/server.ts

```diff
@@ -524,6 +524,10 @@ export class Server {
       if (this.env.packageInfo.buildFlavor !== 'serverless') {
         throw validationError;
       }
+      // Even for serverless mode, we want to throw in CI because that means that the build will include invalid configuration.
+      if (process.env.CI) {
+        throw validationError;
+      }
```

Comment:
This proves to work as expected.

The main downside is (as presented in the risks in the description) that local runs differ from CI runs.

Ideally, the configuration error should occur when running Kibana locally so we can shorten the feedback loop to the dev, and also makes it easier to replicate the error locally. 

To do so, we can add a config key `core.allowStripUnknownsWorkaround` that will be `false` by default and `kibana-controller` sets it to `true`.

I'll submit a new commit with that approach. We can revert it if the reviewers think that an env var is better for this purpose.